tests/components/songpal/test_config_flow.py

```diff
@@ -0,0 +1,249 @@
+    )
+
+
+async def test_flow_ssdp(hass):
+    """Test working ssdp flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_SSDP}, data=SSDP_DATA,
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "init"
+    assert result["description_placeholders"] == {
+        CONF_NAME: FRIENDLY_NAME,
+        CONF_HOST: HOST,
+    }
+    flow = _flow_next(hass, result["flow_id"])
+    assert flow["context"]["unique_id"] == UDN
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={}
+    )
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == FRIENDLY_NAME
+    assert result["data"] == CONF_DATA
+
+
+async def test_flow_user(hass):
+    """Test working user initialized flow."""
+    mocked_device = _create_mocked_device()
+
+    with _patch_config_flow_device(mocked_device):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER},
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "user"
+        assert result["errors"] is None
+        _flow_next(hass, result["flow_id"])
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={CONF_ENDPOINT: ENDPOINT},
+        )
+        assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == MODEL
+        assert result["data"] == {
+            CONF_NAME: MODEL,
+            CONF_ENDPOINT: ENDPOINT,
+        }
+
+    mocked_device.get_supported_methods.assert_called_once()
+    mocked_device.get_interface_information.assert_called_once()
+
+
+async def test_flow_import(hass):
+    """Test working import flow."""
+    mocked_device = _create_mocked_device()
+
+    with _patch_config_flow_device(mocked_device):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=CONF_DATA
+        )
+        assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == FRIENDLY_NAME
+        assert result["data"] == CONF_DATA
+
+    mocked_device.get_supported_methods.assert_called_once()
+    mocked_device.get_interface_information.assert_not_called()
+
+
+def _create_mock_config_entry(hass):
+    MockConfigEntry(domain=DOMAIN, unique_id="uuid:0000", data=CONF_DATA,).add_to_hass(
+        hass
+    )
+
+
+async def test_ssdp_bravia(hass):
+    """Test discovering a bravia TV."""
+    ssdp_data = copy.deepcopy(SSDP_DATA)
+    ssdp_data["X_ScalarWebAPI_DeviceInfo"]["X_ScalarWebAPI_ServiceList"][
+        "X_ScalarWebAPI_ServiceType"
+    ].append("videoScreen")
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_SSDP}, data=ssdp_data,
+    )
+    assert result["type"] == RESULT_TYPE_ABORT
+    assert result["reason"] == "not_songpal_device"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)